working/augmentation-libraries/feature-specification.md

```diff
@@ -1040,136 +810,40 @@ extension type A(int b) {
-
-In all relevant cases, if the augmented member is an instance member, it is
-invoked with the same value for `this`.
-
-Assume that the identifier `augmented` occurs in a source location where no
-enclosing declaration is augmenting. In this case, the identifier is taken
-to be a reference to a declaration which is in scope.
-
-*In other words, `augmented` is just a normal identifier when it occurs
-anywhere other than inside an augmenting declaration.*
-
-*Note that, for example, `augmented()` is an invocation of the augmented
-function or method when it occurs in an augmenting function or method
-declaration. (In the latter case, the augmenting method declaration must
-occur inside an augmenting type-introducing declaration, e.g., an
-augmenting class or mixin declaration). This is also true if `augmented()`
-occurs inside a local function declaration inside the body of that function
-or method declaration. We could say that `augmented` is a contextual
-reserved word because it is unable to refer to a declaration in scope when
-it occurs inside an augmenting declaration, it always has the special
-meaning which is associated with augmentations.*
-
-A compile-time error occurs if a declaration with the basename `augmented`
-occurs in a location where any enclosing declaration is augmenting. *This
-error is applicable to all such declarations, e.g., local functions, local
-variables, parameters, and type parameters.*
-
-Consider a non-augmenting member declaration _Dm_ that occurs inside an
-augmenting type declaration _Dt_. A compile-time error occurs if the
-identifier `augmented` occurs in _Dm_.
-
-*For example, inside `augment class C` we could have a declaration like
-`void f() {...augmented()...}`. This is an error because the outer
-`augment` forces the meaning of `augmented` to be about augmentation in the
-entire scope, but the method declaration is introductory and hence there is
-no earlier declaration to augment.*
+*In practice, most code generators use the [analyzer package][] to introspect
+over code. Code generators introspecting on the _syntax_ of some code likely
+want to see the metadata for each syntactic declaration separately. Code
+generators introspecting over the resolved semantic model of the code (which is
+more common) probably want to see the metadata of the introductory declaration
+and all augmentations appended into a single list of metadata accessible from
+the combined declaration.*
+
+[analyzer package]: https://pub.dev/packages/analyzer
 
 ## Dynamic semantics
 
 The application of augmentation declarations to an augmented declaration
 produces something that looks and behaves like a single declaration: It has
-a single name, a single type or function signature, and it’s what all
+a single name, a single type or function signature, and it's what all
 references to the *name* refers to inside and outside of the library.
 
 Unlike before, that single *semantic declaration* now consists of multiple
```

Comment:
I can't comment on line 896, but I think it would be a good idea to replace non-augmenting by introductory, for consistency.

Reply:
Line 928 ends in "the *C* doe". That sentence needs to be revised and completed. Also, the occurrence of `abstract` (in code font) in a description of an "abstract method" is confusing: The method is abstract, but it can't contain the keyword `abstract`.

Reply:
I'll be honest, I'm not even really sure what this "Dynamic Semantics" section is all about. As far as I'm concerned, all we need to do is specified how augmentations are applied and then the resulting declaration's dynamic semantics are exactly as if you had written everything in one declaration (modulo scoping across part files, of course).

This looks like @lrhn's writing style. (@kallentu and I were coincidentally discussing today how we can tell who wrote each part of a spec 😄 ). @lrhn, do you know what this is supposed to be completed with? Do we need this section at all?